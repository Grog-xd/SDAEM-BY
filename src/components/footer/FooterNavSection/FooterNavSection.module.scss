@import "src/style/mixins";

nav{
  display: flex;
  justify-content: space-between;
  width: 70%;
  .filterNav{
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    height: 90px;
    min-width: 559px;
  }
}
.footerLink{
  @include text(14px, 400, 17px, #8291A3);
  margin-top: 4px;
  margin-left: 15px;
  margin-right: 15px;
  cursor: pointer;
  a{
    @include text(14px, 400, 17px, #8291A3);
    &:hover{
      color: #664EF9;
    }
  }
  b{
    margin-top: 21px;
    @include text(14px, 700, 17px, #1E2123)
  }
}
