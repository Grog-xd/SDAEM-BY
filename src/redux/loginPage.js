import {createSlice} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, setBookMarkActive, registerUser} = loginPage.actions

