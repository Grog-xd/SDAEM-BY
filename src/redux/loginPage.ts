import {createSlice, PayloadAction} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'
import {profilesArr} from '../types/types';

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action:PayloadAction<profilesArr>){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action:PayloadAction<any>){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, registerUser} = loginPage.actions

