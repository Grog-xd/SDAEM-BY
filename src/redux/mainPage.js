import {createSlice} from '@reduxjs/toolkit';

const main = createSlice({
    name: 'main',
    initialState:{
        tabs:[{id:'flats', value:'Квартиры'}, {id:'cottages', value:'Котеджы и усадьбы'}, {id:'baths', value:'Бани и сауны'},  {id:'cars', value:'Авто на прокат'}],

        cityOption:[{id:1,value:'Минск'},{id:2,value:'Гомель'},{id:3,value:'Брест'},{id:4,value:'Витебск'},{id:5,value:'Гродно'}, {id:6,value:'Могилев'}],
        cityCurrentValue:'',

        roomsOption: [{id:1,value:1},{id:2,value:2},{id:3,value:3},{id:4,value:4},{id:5,value:5}],
        roomsCurrentValue:'',

        metroOption:[{id:1,value:'Грушевка'},{id:2,value:'Восход'},{id:3,value:'Пушкинская'},{id:4,value:'Партизанская'}],
        metroCurrentValue:'',

        districtOption:[{id:1,value:'Шабаны'},{id:2,value:'Запад'},{id:3,value:'Восток'},{id:4,value:'Малиновка'}],
        districtCurrentValue: '',

        productsLimit: 6,
        currentPage: 1,

        sortedOptions:[{id:1, value:'Сначала дешевые'}, {id:2, value: 'Сначала дорогие'}],
        sortedValue:'',


        bedSumOptions:[{id:1, value:'1'}, {id:2, value:'2'}, {id:3, value:'3'}, {id:4, value:'4'}, {id:5, value:'5'}],
        bedSumCurrentValue:'',

        checkboxsMoreOption:[{id:1, value:'Газовая плита', bol:false},  {id:2, value:'Духовка', bol:false}, {id:3, value:'Кофеварка', bol:false}, {id:4, value:'Микроволновая печь ', bol:false}, {id:5, value:'Посуда', bol:false}, {id:6, value:'Посудомоечная машина', bol:false}],

        minCost:'',
        maxCost:'',


        productsLen: 0,
        sortedProducts:[],
        products: [],
    },
    reducers: {
        fetchProducts(state, action){
            state.products = action.payload
        },
        setBedSum(state, action){
            state.bedSumCurrentValue = action.payload
        },

        checkboxHandler(state, action){
            state.checkboxsMoreOption.forEach(obj =>{
                if(obj.value.includes(action.payload)){
                    obj.bol = !obj.bol

                }
            })
        },

        likeHandler(state, action){
            state.products.forEach(obj=>{
                Object.keys(obj).forEach(key=>{
                    if(key === action.payload.type){
                        obj[key].forEach(e=>{
                            if (e.id === action.payload.id){
                                e.like = !e.like
                            }
                        })
                    }
                })
            })
        },
        setCity(state, action){
            state.cityCurrentValue = action.payload

        },
        setRooms(state, action){
            state.roomsCurrentValue = action.payload
        },
        setMinCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.minCost = action.payload
        },
        setMaxCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.maxCost = action.payload
        },
        setDistrict(state, action){
            state.districtCurrentValue = action.payload
        },
        setMetro(state, action){
            state.metroCurrentValue = action.payload
        },
        setCurrentPage(state, action){
            state.currentPage = action.payload
            window.scrollTo(0, 0)
        },
        setSortedValue(state, action){
            state.sortedValue = action.payload
        },

        setLimit(state, action){
            state.productsLimit = action.payload
        },
        resetFilter(state){
            state.sortedValue = ''
            state.roomsCurrentValue = ''
            state.minCost = ''
            state.maxCost = ''
            state.districtCurrentValue = ''
            state.metroCurrentValue = ''
            state.bedSumCurrentValue = ''
            state.checkboxsMoreOption.forEach(e=>{
                e.bol = false
            })
        },

        mainFilter(state, action){
            // Условия для высчитывания стоимости
            while(state.minCost > state.maxCost ){
                state.maxCost = +state.maxCost + 100
            }
            if(state.minCost === ''){
                state.minCost=0
            }
            if(state.maxCost === ''){
                state.maxCost = 1000
            }


            //Основная фильтрация
            let sortedRes= []
            state.products.forEach(obj =>{
                if (action.payload){
                    Object.keys(obj).forEach(key =>{
                        if(action.payload === key){
                            obj[key].forEach(e=>{
                                if(!state.roomsCurrentValue){
                                    if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                        sortedRes.push(e)
                                    }
                                } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            })
                        }
                    })
                } else{
                    Object.keys(obj).forEach(key =>{
                        obj[key].forEach(e=>{
                            if(!state.roomsCurrentValue){
                                if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                sortedRes.push(e)
                            }
                        })
                    })
                }
            })
            state.productsLen = sortedRes.length

            // Сортировка массива
            switch (state.sortedValue) {
            case 'Сначала дешевые':
                sortedRes.sort((a, b)=>{
                    return a.cost - b.cost
                })
                break
            case 'Сначала дорогие':
                sortedRes.sort((a, b)=>{
                    return b.cost - a.cost
                })
                break
            default:
                break
            }

            // Фильтрация по страницам
            let res = []
            let test = 0
            for (let i = 0; test < sortedRes.length; i++) {
                let obj = []
                for (let j = 0; j< state.productsLimit; j++) {
                    if(sortedRes[test]){
                        obj.push(sortedRes[test])
                        test++
                    }

                }
                res.push(obj)
            }
            state.sortedProducts = res
            state.currentPage = 1
        },

    },
})

export default main.reducer
export const {setCity, setRooms, setMinCost, setMaxCost, mainFilter, setDistrict, setMetro, resetFilter, setCurrentPage, setSortedValue, setLimit, likeHandler, setBedSum, checkboxHandler, fetchProducts} = main.actions
