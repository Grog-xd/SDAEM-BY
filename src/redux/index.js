import {combineReducers} from "redux";
import {configureStore} from "@reduxjs/toolkit";
import mainPage from "./mainPage";
import newsPage from "./newsPage";
import loginPage from "./loginPage";

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage
})

export const store = configureStore({
    reducer: rootReducer
})
