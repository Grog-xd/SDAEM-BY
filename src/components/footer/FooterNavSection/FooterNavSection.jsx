import React, {memo} from 'react';
import {Link, NavLink} from 'react-router-dom';

import {useDispatch, useSelector} from 'react-redux';

import {setCity} from '../../../redux/mainPage';

import classes from './FooterNavSection.module.scss';

const FooterNavSection = memo(() => {
    const dispatch = useDispatch()

    const {cityOption, tabs}= useSelector(state => state.main)
    let navArr = [{link:'/news', value:'Новости'}, {link:'/404', value:'Размещение и тарифы'}, {link:'/404', value:'Объявления на карте'},{link:'/contacts', value:'Контакты'}]


    return (
        <nav>
            <ul className={classes.filterNav}>
                {tabs.map(tab =>
                    <li key={tab.id}><Link to={`/catalog/${tab.id}`}><b>{tab.value}</b></Link></li>
                )}
                {cityOption.map(city =>
                    <li key={city.id}><Link  onClick={()=> dispatch(setCity(city.value))} to={'/catalog/flats'}>Квартиры в {city.value}е</Link></li>
                )}
            </ul>
            <ul className={classes.pageNav}>
                {navArr.map(nav=>
                    <li key={nav.value}>
                        <NavLink to={nav.link}>{nav.value}</NavLink>
                    </li>
                )}
            </ul>
        </nav>
    );
});

export default FooterNavSection;
