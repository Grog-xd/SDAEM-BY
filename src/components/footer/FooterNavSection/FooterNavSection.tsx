import {FC} from 'react';
import {Link, NavLink} from 'react-router-dom';

import {useDispatch, useSelector} from 'react-redux';

import {setCity} from '../../../redux/mainPage';

import {mainRedux} from '../../../types/types';

import classes from './FooterNavSection.module.scss';

const FooterNavSection:FC = () => {
    const dispatch = useDispatch()

    const {cityOption, tabs}= useSelector((state:mainRedux) => state.main)
    let navArr = [
        {link:'/news', value:'Новости'}, 
        {link:'/404', value:'Размещение и тарифы'}, 
        {link:'/404', value:'Объявления на карте'},
        {link:'/contacts', value:'Контакты'}
    ]


    return (
        <nav data-testid={'footer-nav-section'}>
            <ul className={classes.filterNav}>
                {tabs.map(tab =>
                    <li className={classes.footerLink} key={tab.id}><Link to={`/catalog/${tab.id}`}><b>{tab.value}</b></Link></li>,
                )}
                {cityOption.map(city =>
                    <li className={classes.footerLink} key={city.id}><Link  onClick={()=> dispatch(setCity(city.value))} to={'/catalog/flats'}>Квартиры в {city.value}е</Link></li>,
                )}
            </ul>
            <ul className={classes.pageNav}>
                {navArr.map(nav=>
                    <li className={classes.footerLink} key={nav.value}>
                        <NavLink to={nav.link}>{nav.value}</NavLink>
                    </li>,
                )}
            </ul>
        </nav>
    );
};

export default FooterNavSection;
