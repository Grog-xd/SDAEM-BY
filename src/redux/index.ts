import {combineReducers} from 'redux';
import {configureStore} from '@reduxjs/toolkit';
import {useDispatch} from 'react-redux';

import mainPage from './mainPage';
import newsPage from './newsPage';
import loginPage from './loginPage';

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage,
})

export const store = configureStore({
    reducer: rootReducer,
})


export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch
