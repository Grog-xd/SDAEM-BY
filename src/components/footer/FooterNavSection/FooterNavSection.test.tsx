import {render, screen} from '@testing-library/react';
import {Provider} from 'react-redux';

import {MemoryRouter} from 'react-router-dom';

import {store} from '../../../redux';

import FooterNavSection from './FooterNavSection';

describe('footer nav section tests', () =>{
    test('snapshot test', ()=>{
        render(
            <Provider store={store} >
                <MemoryRouter>
                    <FooterNavSection />
                </MemoryRouter>
            </Provider>,
        )
        const section = screen.getByTestId('footer-nav-section')
        expect(section).toMatchSnapshot();
    })
})
