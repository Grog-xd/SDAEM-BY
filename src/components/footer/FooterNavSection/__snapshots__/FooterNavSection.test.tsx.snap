// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer nav section tests snapshot test 1`] = `
<nav
  data-testid="footer-nav-section"
>
  <ul
    class="filterNav"
  >
    <li
      class="footerLink"
    >
      <a
        href="/catalog/flats"
      >
        <b>
          Квартиры
        </b>
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/cottages"
      >
        <b>
          Котеджы и усадьбы
        </b>
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/baths"
      >
        <b>
          Бани и сауны
        </b>
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/cars"
      >
        <b>
          Авто на прокат
        </b>
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/flats"
      >
        Квартиры в 
        Минск
        е
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/flats"
      >
        Квартиры в 
        Гомель
        е
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/flats"
      >
        Квартиры в 
        Брест
        е
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/flats"
      >
        Квартиры в 
        Витебск
        е
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/flats"
      >
        Квартиры в 
        Гродно
        е
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        href="/catalog/flats"
      >
        Квартиры в 
        Могилев
        е
      </a>
    </li>
  </ul>
  <ul
    class="pageNav"
  >
    <li
      class="footerLink"
    >
      <a
        class=""
        href="/news"
      >
        Новости
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        class=""
        href="/404"
      >
        Размещение и тарифы
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        class=""
        href="/404"
      >
        Объявления на карте
      </a>
    </li>
    <li
      class="footerLink"
    >
      <a
        class=""
        href="/contacts"
      >
        Контакты
      </a>
    </li>
  </ul>
</nav>
`;
